### PR TITLE
Container operators

### PR DESCRIPTION
**Pull request by saeedashrraf**

### CODE REVIEW

File: ivy/container/base.py [hunk 1]
```diff
@@ -684,8 +684,20 @@ def multi_map(
                     )
                     if ret:
                         return_dict[key] = ret
-                elif isinstance(value0, (list, tuple)) and map_nests:
-                    ret = ivy.nested_multi_map(lambda x, _: func(x, None), values)
+                elif any(isinstance(x, (list, tuple)) for x in values) and map_nests:
+                    to_ivy = False
+                    for value in values:
```
Comment (djl11) [inline ivy/container/base.py:689]: could you add comments explaining why this is needed? I would have thought the function being mapped itself converts things to `ivy`? Why do we need to hard code this here you think?

File: ivy/container/container.py [hunk 2]
```diff
@@ -93,15 +93,17 @@ def __pos__(self):
         return self
 
     def __neg__(self):
-        return self.map(lambda x, kc: -x)
+        return self.map(lambda x, kc: -x, map_sequences=True)
```
Comment (djl11) [inline ivy/container/container.py:96]: this is fine, but I think we should generalize `self.map` to also handle `map_nests` argument right? Maybe let's do this is another PR though.

**Review by djl11 — CHANGES_REQUESTED** [review]
small request for comments explaining the `to_ivy` logic. Otherwise looks good. I didn't dive too deep into all the nested implementations, but provided the tests are passing and you are happy that the behavior is as expected - then I'm happy!

File: ivy/container/base.py (hunk 1 above)
Comment (saeedashrraf) [reply ivy/container/base.py:689]: multi_map handles leaves that contain lists or tuples by delegating the work to the nested_multi_map function which by default always returns ivy arrays regardless of the function being mapped. However, for operator functions, if the leaves only contain numbers and not ivy arrays, nested_multi_map should be instructed to return number values and not ivy arrays through the to_ivy variable.

File: ivy/container/container.py (hunk 2 above)
Comment (saeedashrraf) [reply ivy/container/container.py:96]: Currently, `self.map` function can handle `map_nests` with a different variable name `map_sequences` . Do you mean to set `map_sequences` as True by default or what does generalizing `self.map` refer to?
